glean-core/src/upload/mod.rs

```diff
@@ -276,6 +279,64 @@ impl PingUploadManager {
         }
     }
 
+    /// Enqueue pings that might have been cached.
+    ///
+    /// The size of the PENDING_PINGS_DIRECTORY directory will be calculated
+    /// (by accumulating each pings size in that directory)
+    /// and in case we extrapolate the quota, defined by the `quota` arg,
+    /// outstanding pings get deleted and are not enqueued.
+    ///
+    /// The size of the DELETION_REQUEST_PINGS_DIRECTORY will not be calculated
+    /// and no deletion-request pings will be deleted. Deletion request pings
+    /// are not very common and usually don't contain any data,
+    /// we don't expect that directory to ever reach quota.
+    /// Most importantly, we don't want to ever delete deletion-request pings.
+    ///
+    /// Arguments
+    ///
+    /// * `glean` - The Glean object holding the database.
+    /// * `quota` - The quota, in bytes, for the size of the pending pings directory.
+    fn enqueue_cached_pings(&self, glean: &Glean, quota: usize) {
```

Comment:
If I understand this logic right, we essentialy try to do this whenever someone asks for a ping file to be uploaded, but we only read "cached pings" from the directories at startup.
So we effectively clean up the directory only once on first request.
Correct?

Reply:
Correct, especially since we drain the `cached_pings` vector the first time we read it and never write to it again.